shared/override-d.ts/react-redux/index.d.ts

```diff
@@ -0,0 +1,648 @@
+      HandleThunkActionCreator<A7>,
+      HandleThunkActionCreator<A8>
+    ]
+  : TDispatchProps extends [infer A1, infer A2, infer A3, infer A4, infer A5, infer A6, infer A7]
+  ? [
+      HandleThunkActionCreator<A1>,
+      HandleThunkActionCreator<A2>,
+      HandleThunkActionCreator<A3>,
+      HandleThunkActionCreator<A4>,
+      HandleThunkActionCreator<A5>,
+      HandleThunkActionCreator<A6>,
+      HandleThunkActionCreator<A7>
+    ]
+  : TDispatchProps extends [infer A1, infer A2, infer A3, infer A4, infer A5, infer A6]
+  ? [
+      HandleThunkActionCreator<A1>,
+      HandleThunkActionCreator<A2>,
+      HandleThunkActionCreator<A3>,
+      HandleThunkActionCreator<A4>,
+      HandleThunkActionCreator<A5>,
+      HandleThunkActionCreator<A6>
+    ]
+  : TDispatchProps extends [infer A1, infer A2, infer A3, infer A4, infer A5]
+  ? [
+      HandleThunkActionCreator<A1>,
+      HandleThunkActionCreator<A2>,
+      HandleThunkActionCreator<A3>,
+      HandleThunkActionCreator<A4>,
+      HandleThunkActionCreator<A5>
+    ]
+  : TDispatchProps extends [infer A1, infer A2, infer A3, infer A4]
+  ? [
+      HandleThunkActionCreator<A1>,
+      HandleThunkActionCreator<A2>,
+      HandleThunkActionCreator<A3>,
+      HandleThunkActionCreator<A4>
+    ]
+  : TDispatchProps extends [infer A1, infer A2, infer A3]
+  ? [HandleThunkActionCreator<A1>, HandleThunkActionCreator<A2>, HandleThunkActionCreator<A3>]
+  : TDispatchProps extends [infer A1, infer A2]
+  ? [HandleThunkActionCreator<A1>, HandleThunkActionCreator<A2>]
+  : TDispatchProps extends [infer A1]
+  ? [HandleThunkActionCreator<A1>]
+  : TDispatchProps extends Array<infer A>
+  ? Array<HandleThunkActionCreator<A>>
+  : TDispatchProps extends ReadonlyArray<infer A>
+  ? ReadonlyArray<HandleThunkActionCreator<A>>
+  : never
+
+/**
+ * Connects a React component to a Redux store.
+ *
+ * - Without arguments, just wraps the component, without changing the behavior / props
+ *
+ * - If 2 params are passed (3rd param, mergeProps, is skipped), default behavior
+ * is to override ownProps (as stated in the docs), so what remains is everything that's
+ * not a state or dispatch prop
+ *
+ * - When 3rd param is passed, we don't know if ownProps propagate and whether they
+ * should be valid component props, because it depends on mergeProps implementation.
+ * As such, it is the user's responsibility to extend ownProps interface from state or
+ * dispatch props or both when applicable
+ *
+ * @param mapStateToProps
+ * @param mapDispatchToProps
+ * @param mergeProps
+ * @param options
+ */
+
+// KB added
+export type ConnectedComponentType<TMergedProps, TOwnProps> = <C extends ComponentType<any>>(
+  component: C
+) => TMergedProps extends React.ComponentProps<C>
+  ? ConnectedComponentClass<C, TOwnProps>
+  : [
+      // just to help you debug what's going on
+      Exclude<keyof GetProps<C>, keyof TMergedProps>,
+      Exclude<keyof GetProps<C>, keyof GetProps<C>>,
```

Comment:
? Isn't this always going to return an empty set?

Also maybe it would be nice to return a string literal to make clear this is "Debug Info"

Reply:
oh sorry this was a copy and paste error, it's supposed to be TMergedProps first param